### PR TITLE
Change free response structure

### PR DESCRIPTION
**Pull request by yuna878**
Change free response structure to keep track of the answers that user upvoted

### CODE REVIEW

File: src/SessionSocket.js [hunk 1]
```diff
@@ -45,8 +45,9 @@ type Answer = {
  */
 type CurrentState = {
   poll: number,
-  results: {}, // {'A': {'text': 'blue', 'count': 1}}
-  answers: {} // id = client id, answer = current choice
+  results: {}, // MC: {'A': {'text': 'blue', 'count': 1}}, FR: {1: {'text': 'blue', 'count': 1}}
+  answers: {}, // id = client id to answer choice for MC and array of answer ids for FR
```
Comment (young-k) [inline src/SessionSocket.js]: for the ID here, it’s unclear what ID you’re referencing — is answers the array you’re talking about?

File: src/SessionSocket.js [hunk 2]
```diff
@@ -220,22 +224,28 @@ export default class SessionSocket {
...
-              // has selected something before
-              nextState.results[prev].count -= 1;
-              if (poll && poll.type === constants.QUESTION_TYPES.FREE_RESPONSE) {
-                  if (nextState.results[prev].count <= 0) {
-                      delete nextState.results[prev];
-                  }
+
+          if (poll.type === constants.QUESTION_TYPES.MULTIPLE_CHOICE) {
+              nextState.answers[answer.googleId] = answer.choice; // update/add response
+              if (prev) { // if truthy
```
Comment (young-k) [inline src/SessionSocket.js]: doesn’t an if statement imply truthiness? maybe put the truthy values you’re checking, to be more clear

File: src/SessionSocket.js [hunk 3]
```diff
@@ -220,22 +224,28 @@ export default class SessionSocket {
...
+              nextState.answers[answer.googleId] = [answer.id];
           }
 
-          const curTally = nextState.results[answer.choice];
-          if (curTally) { // if truthy
-              nextState.results[answer.choice].count += 1;
+          const key = poll.type === constants.QUESTION_TYPES.MULTIPLE_CHOICE
+              ? answer.choice : answer.id;
+
+          if (nextState.results[key]) { // if truthy
```
Comment (young-k) [inline src/SessionSocket.js]: same here

File: src/SessionSocket.js [hunk 4]
```diff
@@ -245,33 +255,23 @@ export default class SessionSocket {
...
+          const curTally = nextState.results[answerId];
           if (curTally) {
-              nextState.results[answer.choice].count += 1;
-          } else {
-              nextState.results[answer.choice] = { text: answer.text, count: 1 };
+              nextState.results[answerId].count += 1;
+              if (nextState.upvotes[googleId]) {
+                  nextState.upvotes[googleId].push(answerId);
+              } else {
+                  nextState.upvotes[googleId] = [answerId];
```
Comment (young-k) [inline src/SessionSocket.js]: nitpicky, but all these variables should be ID instead of Id

File: src/SessionSocket.js [hunk 5]
```diff
@@ -220,22 +224,28 @@ export default class SessionSocket {
 
           const nextState = { ...this.current };
           const prev = nextState.answers[answer.googleId];
```
Comment (young-k) [inline src/SessionSocket.js]: unclear how a variable can be called prev when it’s coming from nextState?

File: src/SessionSocket.js (hunk 1 above)
Comment (yuna878) [reply src/SessionSocket.js]: Not too sure what you are asking?.. The 'array' refers to an array of answer object's id

Comment (young-k) [reply src/SessionSocket.js]: oh i see is this a dictionary with an `id` field and an `array`? can we make explicit types for these?

**Pull request merged**